test/boostTest.cpp

```diff
@@ -86,14 +86,20 @@ int registerTests(
 			fs::directory_iterator(fullpath),
 			fs::directory_iterator()
 		))
-			if (fs::is_directory(entry.path()) || TestCase::isTestFilename(entry.path().filename()))
+		{
+			bool addTest = true;
+			for (auto const& element: entry.path())
+				if (boost::starts_with(element.string(), "_"))
+					addTest = false;
+			if (addTest && (fs::is_directory(entry.path()) || TestCase::isTestFilename(entry.path().filename())))
```

Comment:
I think it is hard to read as one line, maybe splitting like this
```suggestion
			if (
				addTest && 
				(fs::is_directory(entry.path()) || TestCase::isTestFilename(entry.path().filename()))
			)
```